dev-learning/Controllers/CommentsController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using dev_learning.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+
+namespace dev_learning.Controllers
+{
+    [Route("api/[controller]")]
+    public class CommentsController : Controller
+    {
+        private readonly MyDbContext _context;
+
+        public CommentsController(MyDbContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Comments/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Comment>> GetComment(int id)
+        {
+            var comment = await _context.Comments.FindAsync(id);
+
+            if (comment == null)
+            {
+                return NotFound();
+            }
+
+            return comment;
+        }
+
+        // GET: api/Comments/User/5
+        [HttpGet("User/{userId}")]
+        public async Task<List<Comment>> GetUserComments(int userId)
+        {
+            return await _context.Comments.Where(c => c.UserId == userId).ToListAsync();
+        }
+
+        // GET: api/Comments/Subject/5
+        [HttpGet("Subject/{subjectId}")]
+        public async Task<List<Comment>> GetSubjectComments(int subjectId)
+        {
+            return await _context.Comments.Where(c => c.SubjectId == subjectId).ToListAsync();
+        }
+
+        // POST api/Comments
+        [HttpPost]
+        public async Task<ActionResult<Comment>> PostComment([FromBody]Comment comment)
+        {
+            comment.User = _context.Users.FirstOrDefault(x => x.Id == comment.UserId);
+
+            comment.User.Comments.Add(comment);
```

Comment:
What are these `User` `Comments`? So `User` writes a `Comment` under a `Subject` - that's all. But we somehow end up with 3 types of comments - user comments, subject comments and all comments 😄

Reply:
From the frontend perspective, I can only see 2 use cases:

- I want to get all the comments on a specific subject by subjects ID.
- I want to get user comments on assigned subject

> Mokymosi dienos komentarai. Nuorodos į naudotą medžiagą. Prideda darbuotojas.